apps/core/app/[locale]/(default)/login/_components/reset-password-form.tsx

```diff
@@ -0,0 +1,166 @@
+'use client';
+
+import { Button } from '@bigcommerce/components/button';
+import {
+  Field,
+  FieldControl,
+  FieldLabel,
+  FieldMessage,
+  Form,
+  FormSubmit,
+} from '@bigcommerce/components/form';
+import { Input } from '@bigcommerce/components/input';
+import { Message } from '@bigcommerce/components/message';
+import { Loader2 as Spinner } from 'lucide-react';
+import { useTranslations } from 'next-intl';
+import { ChangeEvent, useRef, useState } from 'react';
+import { useFormStatus } from 'react-dom';
+import ReCaptcha from 'react-google-recaptcha';
+
+import { submitResetPasswordForm } from '../_actions/submit-reset-password-form';
+
+interface Props {
+  reCaptchaSettings?: {
+    isEnabledOnStorefront: boolean;
+    siteKey: string;
+  };
+}
+
+interface FormStatus {
+  status: 'success' | 'error';
+  message: string;
+}
+
+const SubmitButton = () => {
+  const { pending } = useFormStatus();
+  const t = useTranslations('Account.SubmitResetPassword');
+
+  return (
+    <Button
+      className="relative w-fit items-center px-8 py-2"
+      data-button
+      disabled={pending}
+      variant="primary"
+    >
+      <>
+        {pending && (
+          <>
+            <span className="absolute z-10 flex h-full w-full items-center justify-center bg-gray-400">
+              <Spinner aria-hidden="true" className="animate-spin" />
+            </span>
+            <span className="sr-only">{t('spinnerText')}</span>
+          </>
+        )}
+        <span aria-hidden={pending}>{t('submitText')}</span>
+      </>
+    </Button>
```

Comment:
Just noticed the same component is used in #665 change password form.

Reply:
Yeah. Actually, now we have around 6 forms with submit button which are different basically only  with their texts. I was thinking about creating abstraction with `text` params that will return `<SubmitButton>` & reuse it across all forms💅